### PR TITLE
Allowing decompression prevention

### PR DESCRIPTION
**Pull request by blockiosaurus**
This PR aims to allow tree creators to manage whether or not a tree is decompressable. This is helpful to prevent heterogenous collections for protocols that can't handle both cNFTs and NFTs. This is accomplished through the following means:

- A `SetDecompressableState` Instruction has been added to toggle whether or not decompression is allowed.
- A new field on the `TreeConfig` called `is_decompressable` which indicates whether or not it is allowed.
  - All old Trees will be decompressable as they were before.
  - All new trees will **not** be decompressable so the decision to decompression will have to be explicit.
- A `DecompressableState` enum that provides a human readable description of whether or not it is enabled.

### CODE REVIEW

File: programs/bubblegum/program/tests/decompression.rs
Comment (danenbm) [inline programs/bubblegum/program/tests/decompression.rs:1]: Nice!  and thank you for splitting this out to a new file as well.

File: clients/js/src/generated/accounts/treeConfig.ts
Comment (danenbm) [inline clients/js/src/generated/accounts/treeConfig.ts:1]: Did you have a JS test to add for this feature?  I'm not saying we need one, but I thought you mentioned offline that you were writing one, so wanted to make sure it was included if you had written it.

**Review by danenbm — CHANGES_REQUESTED** [review]

File: clients/js/src/generated/accounts/treeConfig.ts
[the remark this answers is not in the document]
Comment (blockiosaurus) [reply clients/js/src/generated/accounts/treeConfig.ts:1]: Eh, I think the Umi library is pretty complete, so we can always hold off until someone asks for it.

Comment (blockiosaurus) [reply clients/js/src/generated/accounts/treeConfig.ts:1]: I did not, I was just trying to fix the tests I broke for the redeem and decompress JS tests. Lemme commit those changes.

File: yarn.lock
[the remark this answers is not in the document]
Comment (blockiosaurus) [reply yarn.lock:1]: Whoops!

File: clients/js/src/generated/accounts/treeConfig.ts
[the remark this answers is not in the document]
Comment (danenbm) [reply clients/js/src/generated/accounts/treeConfig.ts:1]: ok per offline discussion, our plan will be for short term we will still regenerate it.

And then we come up with the metric for deciding most people are using now using Umi, to decide when to actually delete JS Solita.

**Review by danenbm — DISMISSED** [review]
Code change looks great!

Per offline discussion, our plan will be for short term we will still regenerate JS Solita but in a follow on PR since its not properly setup right now.

And then we come up with the metric for deciding most people are using now using Umi, to decide when to actually delete JS Solita.

**Review by febo — APPROVED** [review]

**Review by danenbm — APPROVED** [review]

**Pull request merged**